### PR TITLE
As a user i should be able to update my channel details after i publish a story #342

### LINKED ISSUE #342: 🎨 🚀 As a user I should be able to update my channel details after I publish a story
## User stories

As a user I should be able to update my channel details after I publish a story. This will make it easier for me to update my channel details.

---

Quick links: [Summary](#summary) | [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

---

### Description

As a user I should be able to update my channel details after I publish a story. This will make it easier for me to update my channel details.

Currently, after I have published a story, I have to publish the story again, in an attempt to update the details. But even this does not work. 

Therefore, after publishing a story the first time, the publish story button text should change to `Update Channel Details`. And when i click it, the form should open populated with preexisting details. I should be able to update the details with new ones and save (except for the phone number id).

### Acceptance Criteria

1. After I published the story the first time, I should not have to save it again to update the channel details.
2. Any updates made should reflect on the db under `channels/{{channelId}}`
3. Any other fields not updated should remain intact.

### Note

The publish story button runs a distributed counter when a new story is published, which increments the technical ref of a channel, that is `n`. No need to get into the details now, but just as a caution, this should also not change when a channel is updated.

### PR DESCRIPTION
**Pull request by 23po**
# Description

This PR makes it possible for a user to update their channel details after publishing a story.

Fixes #342

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# Video
[screen-capture (4).webm](https://user-images.githubusercontent.com/48943229/226799978-f0e179db-b32c-4a80-a23a-53eaabf1f19c.webm)

 

# How Has This Been Tested?

This has been tested on the development server, and the changes observed to reflect on the db under channels

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by 23po** [comment]
@LemmyMwaura @ReaganCn please review

File: libs/features/convs-mgr/stories/editor/src/lib/pages/story-editor/story-editor.page.ts [hunk 1]
```diff
@@ -20,13 +20,32 @@ import { StoryEditorFrame } from '../../model/story-editor-frame.model';
 import { AddBotToChannelModal } from '../../modals/add-bot-to-channel-modal/add-bot-to-channel.modal';
 
 import { getActiveBlock } from '../../providers/fetch-active-block-component.function';
+import { CommunicationChannel } from '@app/model/convs-mgr/conversations/admin/system';
+import { SharedService } from '../../components/shared-service';
+import { ActiveStoryStore } from '@app/state/convs-mgr/stories';
+import { Story } from '@app/model/convs-mgr/stories/main';
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/editor/src/lib/pages/story-editor/story-editor.page.ts:26]: Check order of imports. Domain imports should come before local imports

File: libs/features/convs-mgr/stories/editor/src/lib/components/shared-service.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+import { CommunicationChannel } from '@app/model/convs-mgr/conversations/admin/system';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SharedService {
+  private _isPublished = new Subject<CommunicationChannel>();
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/editor/src/lib/components/shared-service.ts:9]: Please use BehaviourSubject instead, and set the initial value to false

File: libs/features/convs-mgr/stories/editor/src/lib/providers/manage-channel-story-link.service.ts [hunk 3]
```diff
@@ -35,17 +37,40 @@ export class ManageChannelStoryLinkService
...
+    //   id: existingChannel.id,
+    //   orgId: existingChannel.orgId,
+    //   n: existingChannel.n,
+    //   type: existingChannel.type,
+    //   name: updatedChannel.name,
+    //   description: updatedChannel.description,
+    //   // Add other properties that can be updated here
+    //   phone_number_id: existingChannel.phone_number_id
+    // };
+
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/editor/src/lib/providers/manage-channel-story-link.service.ts:66]: Remove this commented out code.

File: libs/features/convs-mgr/stories/editor/src/lib/providers/manage-channel-story-link.service.ts [hunk 4]
```diff
@@ -90,4 +115,10 @@ export class ManageChannelStoryLinkService
     const channelRepo = this._getChannelRepo(channel);
     return channelRepo.getDocuments(new Query().where("id", "==", channel.id));
   }
+  
+  // public getData(channelId) {
+  //   const channelRepo = this._getChannelRepo(channel);
+  //   return channelRepo.getDocuments(new Query().where("id", "==", channelId));
+  // }
+
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/editor/src/lib/providers/manage-channel-story-link.service.ts:123]: Also this

**Review by ReaganCn — CHANGES_REQUESTED** [review]
Good first attempt. But there are still some changes that you need to address:
- Your code has a bug that will constatly update the details endlessly (endless loop). 
- There are some rendundacies which I have highlighted
- Finally your video needs to show everything from publishing the details when they dont exist to updating them.

**Comment by njoroge-jane** [comment]
@23po Kindly do make the requested changes and notify once done or if you might need more clarification or assistance.

**Comment by 23po** [comment]
> @23po Kindly do make the requested changes and notify once done or if you might need more clarification or assistance.

sure

**Comment by LemmyMwaura** [comment]
Closing this to avoid backlog

**Comment by 23po** [comment]
hi @LemmyMwaura i am working on the issue

**Comment by LemmyMwaura** [comment]
Please communicate if you need assistance.

**Comment by LemmyMwaura** [comment]
Closing this for #473 (opened by the same user)

**Pull request closed without merging**